core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -685,6 +685,8 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
         }
         lastInode.setPinned(currentInodeDirectory.isPinned());
 
+        // Update state while holding the write lock.
+        mInodes.add(lastInode);
         if (!currentInodeDirectory.addChild(lastInode)) {
```

Comment:
Does the inode have to be removed in this block?

Reply:
done